### PR TITLE
Clean up paged result handling.

### PR DESCRIPTION
**Pull request by come-nc**
There is only one paged result API now that PHP 7.2 is long dead.

- [x] Starting by removing empty function calls.
- [x] Pass pagination data to search call directly?
- [x] Get rid of adapter?
- [x] Make sure read calls are not paginated and not logged as paginated
- [x] Properly log request and responses

Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

### CODE REVIEW

File: apps/user_ldap/lib/Access.php [hunk 1]
```diff
@@ -194,15 +193,6 @@ public function readAttribute(string $dn, string $attr, string $filter = 'object
 			$this->logger->debug('LDAP resource not available.', ['app' => 'user_ldap']);
 			return false;
 		}
-		//Cancel possibly running Paged Results operation, otherwise we run in
-		//LDAP protocol errors
-		$this->abandonPagedSearch();
```
Comment (blizzz) [inline apps/user_ldap/lib/Access.php:199]: I am a bit concerned about removing this one here. Ideally, when going through a paged results search, there is no side-reading of LDAP attributes. I would keep it as safe guard nevertheless.

Comment (come-nc) [reply apps/user_ldap/lib/Access.php:199]: This call is only reseting a few properties on the class, like lastCookie and pagedResultSuccessful. These are not used in the case of a read as we do not send LDAP pagination controls with the read.
LDAP controls are specific to each request now, we do not use ldap_set_option for this anymore.

**Review by CarlSchwan — APPROVED** [review]

**Review by blizzz — APPROVED** [review]

**Comment by come-nc** [comment]
CI failure unrelated

**Pull request merged**